src/satellitePropagator/physics/YoshidaIntegrator.h

```diff
@@ -0,0 +1,297 @@
+     * @param accumulator_arg Reference to the
+     * Acceleration::AccelerationAccumulator object to calculate acceleration for
+     * the current time step
+     * @param delta_t_arg Time step to Integrate over
+     */
+    YoshidaIntegrator(Container& container,
+        Acceleration::AccelerationAccumulator<Container>& accumulator_arg,
+        double delta_t_arg)
+        : container(&container)
+        , accumulator(&accumulator_arg)
+        , delta_t(delta_t_arg)
+        , delta_tDiv4(delta_t_arg / 4.){};
+
+    /**
+     * @brief Default destructor
+     *
+     * Destroys the Integrator object
+     */
+    virtual ~YoshidaIntegrator();
+
+    /**
+     * @brief Calculates a complete time step
+     *
+     * Integrator::calculateAcceleration() is called first.
+     * Because the Integrator::calculatePosition() function needs the
+     * Debris::Debris::velocity vector of the last time step of the
+     * Debris::Debris objects of the Container #container it is called
+     * before Integrator::calculateVelocity()
+     *
+     * @param write_time_step if true all calculated acceleration components are writen to a file
+     */
+    void integrate(bool write_time_step = false) const;
+
+    /**
+     * @brief Calculates the new position
+     *
+     * Calculates the Debris::Debris::position vector for all Debris::Debris
+     * objects of the Container #container Uses leapfrog integration
+     * with Debris::Debris::velocity and Debris::Debris::acc_t0
+     *
+     */
+    void calculatePosition(double c) const;
+
+    /**
+     * @brief Calculates the new velocities
+     *
+     * Calculates the Debris::Debris::velocity vector for all Debris::Debris
+     * objects of the Container #container Uses leapfrog integration
+     * with Debris::Debris::acc_t0 and Debris::Debris::acc_t1
+     *
+     */
+    void calculateVelocity(double d) const;
+
+    /**
+     * @brief Calculates te acceleration for the current time step
+     *
+     * Calls the Acceleration::AccelerationAccumulator::applyComponents()
+     * function of the Acceleration::AccelerationAccumulator object #accumulator
+     * After this function is called the Debris::Debris objects of the
+     * Container #container are ready for integrating
+     *
+     * @param write_time_step if true all calculated acceleration components are writen to a file
+     */
+    void calculateAcceleration(bool write_time_step) const;
+
+private:
+    Container*
+        container
+        = nullptr; /**< Reference to the Container object holding the
+             Debris::Debris objects to integrate for*/
+    Acceleration::AccelerationAccumulator<Container>*
+        accumulator
+        = nullptr; /**< Reference to the Acceleration::AccelerationAccumulator
+                   object to calculate acceleration for the current time
+                   step*/
+    double delta_t; /**< Time step to Integrate over */
+    double delta_tDiv4; /**< Time step divided by 4 */
```

Comment:
So reading more about this, I think we actually shouldn't do this. From the papers I am understanding that yoshida does some backward in time time steps as well as forward, not some smaller ones. 

Unfortunately, our current scheme implicitly encodes at which times each step happens and I don't quite get how to reconstruct this. 🤔 In any event it is definitely not just 0.25 dt.

I think for now we can neglect this but it may main that we are making some additional error by evaluating the spherical harmonics and other t dependent forces at the wrong time 🤔

Reply:
For simplicity we could at least write that down / create an issue to look into another method that does not have this drawback or where we aware of the exact evaluation points in time? (maybe in conjunction with the papers I sent you?)

Reply:
-> #7